rollbar-client/src/Rollbar/Client/Item.hs

```diff
@@ -80,28 +80,38 @@ data Item = Item
   , itemServer :: Maybe Server
     -- ^ Data about the server related to this event.
   -- client
-  -- custom
-  -- fingerprint
-  -- title
-  -- uuid
+  , custom :: Maybe Object
+    -- ^ Any arbitrary metadata you want to send.
+  , title :: Maybe Text
+    -- ^ A string that will be used as the title of the Item occurrences will be grouped into.
+  , uuid :: Maybe Text
+    -- ^ Identifies this occurrence.
+  , fingerprint :: Maybe Text
+    -- ^ Controlls how this occurrence should be grouped.
   , itemNotifier :: Notifier
     -- ^ Describes the library used to send this event.
   } deriving (Eq, Show)
 
 instance ToJSON Item where
-  toJSON Item{..} = object
-    [ "data" .= object
-        [ "environment" .= itemEnvironment
-        , "body" .= itemBody
-        , "level" .= itemLevel
-        , "platform" .= itemPlatform
-        , "language" .= itemLanguage
-        , "framework" .= itemFramework
-        , "request" .= itemRequest
-        , "server" .= itemServer
-        , "notifier" .= itemNotifier
-        ]
-    ]
+  toJSON Item{..} =
+    let dataFields =
+          [ "environment" .= itemEnvironment
+          , "body" .= itemBody
+          , "level" .= itemLevel
+          , "platform" .= itemPlatform
+          , "language" .= itemLanguage
+          , "framework" .= itemFramework
+          , "request" .= itemRequest
+          , "server" .= itemServer
+          , "notifier" .= itemNotifier
+          ] ++ catMaybes
+          [ ("custom" .=) <$> custom
+           , ("fingerprint" .=) <$> fingerprint
```

Comment:
Oddly rollbar's API doesn't like fingerprint being set to null but will accept it if it's null. It errors with fingerprint should be a string.